isis/tests/Hyabusa2OncCameraTests.cpp

```diff
@@ -0,0 +1,245 @@
+#include <QDebug>
+
+#include <iomanip>
+#include <iostream>
+
+#include "Hyb2OncCamera.h"
+#include "Camera.h"
+#include "CameraFactory.h"
+#include "IException.h"
+#include "iTime.h"
+#include "Preference.h"
+#include "Pvl.h"
+#include "TestUtilities.h"
+#include "PvlGroup.h"
+#include "Fixtures.h"
+
+#include "gmock/gmock.h"
+
+using namespace std;
+using namespace Isis;
+
+void testCamera(Cube &c, double knownLat, double knownLon,
+                double s1, double l1, 
+                double s2, double l2, 
+                double s3, double l3, 
+                double s4, double l4);
+void testLineSamp(Camera *cam, double sample, double line);
+
+TEST_F(DefaultCube, Hayabusa2OncCameraW1CameraTest) {
+  PvlGroup &kernels = testCube->label()->findObject("IsisCube").findGroup("Kernels");
+  kernels.findKeyword("NaifFrameCode").setValue("-37110");    
+  
+  PvlGroup &inst = testCube->label()->findObject("IsisCube").findGroup("Instrument");
+  inst.findKeyword("InstrumentId").setValue("ONC-W2");
+  inst.findKeyword("SpacecraftName").setValue("HAYABUSA-2");
+  PvlKeyword startcc("SpacecraftClockStartCount", "33322515");
+  PvlKeyword stopcc("SpaceCraftClockStopCount", "33322515");
+  PvlKeyword binning("Binning", "1");
+  inst += startcc;
+  inst += stopcc; 
+  inst += binning; 
+
+  PvlObject &naifKeywords = testCube->label()->findObject("NaifKeywords");
+  
+  json nk; 
+  nk["INS-37110_FOCAL_LENGTH"] = 10.44;
+  nk["INS-37110_PIXEL_PITCH"] = 0.013;
+  nk["INS-37110_TRANSX"] = {0.0, 0.013, 0.0};
+  nk["INS-37110_TRANSY"] = {0.0, 0.0, 0.013};
+  nk["INS-37110_ITRANSS"] = {0.0, 76.923076923077, 0.0};
+  nk["INS-37110_ITRANSL"] = {0.0, 0.0, 76.923076923077};
+  nk["INS-37110_BORESIGHT_LINE"] = 490.5;
+  nk["INS-37110_BORESIGHT_SAMPLE"] = 512.5;
+  nk["INS-37110_OD_K"] = {1.014, 2.933e-07, -1.384e-13};
+  nk["BODY499_RADII"] = {3396.19, 3396.19, 3376.2};
+  nk["CLOCK_ET-37_33322515_COMPUTED"] = "8ed6ae8930f3bd41";
+  nk["BODY_CODE"] = 499;
+  nk["BODY_FRAME_CODE"] = 10014; 
+  PvlObject newNaifKeywords("NaifKeywords", nk);
+  naifKeywords = newNaifKeywords; 
+
+  QString fileName = testCube->fileName();
+  // need to remove old camera pointer 
+  delete testCube;
+  // This is now a Hayabusa cube
+  testCube = new Cube(fileName, "rw");
```

Comment:
Is it possible to put the bulk of this into a fixture that resides solely in this file?